pkg/controller/storagecapabilities.go

```diff
@@ -53,13 +56,49 @@ var CapabilitiesByProvisionerKey = map[string][]StorageCapabilities{
 	"csi.trident.netapp.io/ontap-san": {{AccessMode: v1.ReadWriteOnce, VolumeMode: v1.PersistentVolumeBlock}},
 }
 
-// Get finds and returns a predefined StorageCapabilities for a given StorageClass
-func Get(sc *storagev1.StorageClass) ([]StorageCapabilities, bool) {
+// GetStorageCapabilities finds and returns a predefined StorageCapabilities for a given StorageClass
+func GetStorageCapabilities(cl client.Client, sc *storagev1.StorageClass) ([]StorageCapabilities, bool) {
 	provisionerKey := storageProvisionerKey(sc)
+	if provisionerKey == "kubernetes.io/no-provisioner" {
+		return capabilitiesForNoProvisioner(cl, sc)
+	}
 	capabilities, found := CapabilitiesByProvisionerKey[provisionerKey]
 	return capabilities, found
 }
 
+func capabilitiesForNoProvisioner(cl client.Client, sc *storagev1.StorageClass) ([]StorageCapabilities, bool) {
+	pvs := &v1.PersistentVolumeList{}
+	err := cl.List(context.TODO(), pvs)
+	if err != nil {
+		return []StorageCapabilities{}, false
+	}
+	capabilities := []StorageCapabilities{}
+	for _, pv := range pvs.Items {
```

Comment:
We need a pvc, and then we can provide the self-configuring magic. I am not sure what think about this.  But I think there is no other easy way to detect capabilities.

Reply:
Another question. Do we need to scan all PVs, is there any shortcut possible?

Reply:
Worth noting that no-provisioner storage class implies nothing provisions these PVs automatically, you have to create PVs ahead of time to use them, so if they don't exist the PVCs are going to remain in pending state forever (even if correctly defined).

Reply:
Right!